### ISSUE TITLE
Dependabot groups does not include all dependencies

### ISSUE THREAD

**Issue by v3gard** [dependabot/dependabot-core#7530]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

npm

### Package manager version

8.3.1

### Language version

16.14.0

### Manifest location and content before the Dependabot update

/package.json

### dependabot.yml content

```yml
version: 2
updates:
    - package-ecosystem: "npm"
      directory: "/"
      schedule:
          interval: "weekly"
          day: "sunday"
      open-pull-requests-limit: 10
      reviewers:
          - "equinor/kpd-team"
      groups:
          ag-grid:
              patterns:
                  - "*ag-grid-*"
                  - "*fusion-framework-module-ag-grid"
      ignore:
          # Ignore react/react-dom related major upgrades until Fusion support major version upgrade.
          - dependency-name: "react-dom"
            update-types: ["version-update:semver-major"]
          - dependency-name: "@types/react-dom"
            update-types: ["version-update:semver-major"]
          - dependency-name: "react"
            update-types: ["version-update:semver-major"]
          - dependency-name: "@types/react"
            update-types: ["version-update:semver-major"]
```

### What you expected to see, versus what you actually saw

 I expected all AG grid packges to be updated at the same time. Instead, some were left out, e.g.:
 
 * `@ag-grid-community/core`
 * `@ag-grid-community/react`

![image](https://github.com/dependabot/dependabot-core/assets/343078/b8eb2d5f-a961-4b88-807c-c67532719229)

**Comment by v3gard** [comment]
For reference, here's what the actual PR looked like:
![image](https://github.com/dependabot/dependabot-core/assets/343078/fc56ecde-a4e6-4eb2-8ea1-c3834bf4af3d)

Also note that at the time of writing, [@ag-grid-community/core](https://www.npmjs.com/package/@ag-grid-community/core) 
was published 3 days ago.

![image](https://github.com/dependabot/dependabot-core/assets/343078/3b8f60e0-f1a9-4571-b259-777c61a11e8f)

**Comment by v3gard** [comment]
This might be related to #7471

**Comment by Nishnha** [comment]
Hi I attempted to [recreate](https://github.com/dark-dependabot/ag-grid-grouped-update/pull/1) this issue and noticed that some of the same packages indeed failed to update.

I took a look at the update logs by going to the insights tab > Dependency graph > Dependabot > and clicking on the "last checked X minutes ago" link.

The logs show that some of these updates are not possible:

```
updater | 2023/07/07 15:41:17 INFO <job_689802642> Checking if @ag-grid-community/react 29.3.1 needs updating
  proxy | 2023/07/07 15:41:17 [028] GET https://registry.npmjs.org:443/@ag-grid-community%2Freact
  proxy | 2023/07/07 15:41:17 [028] 200 https://registry.npmjs.org:443/@ag-grid-community%2Freact
  proxy | 2023/07/07 15:41:17 [030] GET https://registry.npmjs.org:443/@ag-grid-community%2Freact/30.0.4
  proxy | 2023/07/07 15:41:17 [030] 200 https://registry.npmjs.org:443/@ag-grid-community%2Freact/30.0.4
updater | 2023/07/07 15:41:17 INFO <job_689802642> Latest version is 30.0.4
  proxy | 2023/07/07 15:41:18 [032] GET https://registry.npmjs.org:443/@ag-grid-community%2freact
  proxy | 2023/07/07 15:41:18 [032] 200 https://registry.npmjs.org:443/@ag-grid-community%2freact
  proxy | 2023/07/07 15:41:18 [034] GET https://registry.npmjs.org:443/react
  proxy | 2023/07/07 15:41:18 [034] 200 https://registry.npmjs.org:443/react
  proxy | 2023/07/07 15:41:19 [036] GET https://registry.npmjs.org:443/react-dom
  proxy | 2023/07/07 15:41:19 [036] 200 https://registry.npmjs.org:443/react-dom
updater | 2023/07/07 15:41:19 INFO <job_689802642> Requirements to unlock update_not_possible
```

Could you check to see if your logs show similar errors?
Update not possible errors can be caused by conflicting dependencies - but the update may be possible in the future as maintainers resolve these conflicts and publish new versions of the package. You may also have luck locally updating the package with npm.

Please let us know what you find! If the issue is not related to an update not possible or similar error we can take another look :)

**Comment by v3gard** [comment]
🤷 I noticed that `@ag-grid-community/core` could be removed as an explicit dependency, as it was installed in `node_modules` as a transitive dependency. I did the following:

> `npm uninstall "@ag-grid-community/core"`

Which resulted in this change set:
![image](https://github.com/dependabot/dependabot-core/assets/343078/6bc311a0-1e0c-46d5-933d-d323b065971d)

After pushing the changes to the main  branch, the problem was solved. This made Dependabot re-run the grouping feature for my 5 day old pull request and the problem was solved.

![image](https://github.com/dependabot/dependabot-core/assets/343078/496b60df-8a56-4f57-8c77-2e6410bf10c6)

**Comment by jakecoffman** [comment]
We're seeing another issue from multiple customers where `@angular` grouped updates are having a different variant of the same problem. I've recreated that here: https://github.com/dsp-testing/angular-grouped-update

This is due to peer dependencies on `@angular/core` causing the update to fail, but the peers themselves are able to be updated.

**Comment by jakecoffman** [comment]
This should be fixed now by #7561.
